salt/states/flatpak.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+'''
+Management of flatpak packages
+==============================
+Allows the installation and uninstallation of flatpak packages.
+
+.. versionadded:: Neon
+'''
+from __future__ import absolute_import, print_function, unicode_literals
+
+import salt.utils.path
+
+__virtualname__ = 'flatpak'
+
+
+def __virtual__():
+    if salt.utils.path.which('flatpak'):
+        return __virtualname__
+
+    return (False, 'The flatpak state module cannot be loaded: the "flatpak" binary is not in the path.')
+
+
+def installed(location, name):
+    '''
+    Ensure that the named package is installed
+    location
+        The location or remote to install the flatpak from.
+    name
+        The name of the package or runtime
+    '''
+    ret = {'name': name,
+           'changes': {},
+           'pchanges': {},
```

Comment:
We're not using pchanges anymore. Changes that would happen if `test=True` should just go in `changes` with `result=None`

This needs to be applied throughout the state module